### PR TITLE
[f] 555a - cpdb bot features

### PR DESCRIPTION
**Pull request by giang271291**
Description
--
- Follow user
- Parse quoted_status_id_str, in_reply_to_status_id(_str)
- Set cause for UserHandlerError
- Ignore Tweepy stream error
- Set logging level to INFO
- Fix match_tweet when bot run on user_stream

### CODE REVIEW

File: responsebot/models.py [hunk 1]
```diff
@@ -72,19 +78,31 @@ def __init__(self, track=[], follow=[]):
...
         :param tweet: The tweet in question
         :type tweet: :class:`~responsebot.models.Tweet`
         :return: True if matched, False otherwise
         """
+        if user_stream:
+            if len(self.track) > 0:
+                return self.is_tweet_match_track(tweet)
+
+            return True
+        else:
```
Comment (tklarryonline) [inline responsebot/models.py]: `else` is not needed here.

File: responsebot/responsebot.py [hunk 2]
```diff
@@ -49,20 +49,15 @@ def start(self):
             sys.exit()
 
         try:
-            client = auth_utils.auth(
-                consumer_key=self.config.get('consumer_key'),
-                consumer_secret=self.config.get('consumer_secret'),
-                token_key=self.config.get('token_key'),
-                token_secret=self.config.get('token_secret'),
-            )
+            client = auth_utils.auth(self.config)
```
Comment (tklarryonline) [inline responsebot/responsebot.py]: 👍

File: responsebot/responsebot_stream.py [hunk 3]
```diff
@@ -52,3 +51,7 @@ def start(self, retry_limit=None):
                 if 'Stream object already connected!' in e.reason or \
                         'Wrong number of locations points' in e.reason:
                     raise APIError(e.reason)
+            except AttributeError as e:
+                # Known Tweepy's issue https://github.com/tweepy/tweepy/issues/576
+                if '\'NoneType\' object has no attribute \'strip\'' in str(e):
```
Comment (tklarryonline) [inline responsebot/responsebot_stream.py]: You can use double quotes here to do single quotes in between the string.

File: responsebot/utils/auth_utils.py [hunk 4]
```diff
@@ -12,21 +12,19 @@
...
+    :type config: :class:`~responsebot.utils.config_utils.ResponseBotConfig`
     :return: client instance to execute twitter action
     :rtype: :class:`~responsebot.responsebot_client.ResponseBotClient`
     :raises: :class:`~responsebot.common.exceptions.AuthenticationError`: If failed to authenticate
     :raises: :class:`~responsebot.common.exceptions.APIQuotaError`: If API call rate reached limit
     """
-    auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
-    auth.set_access_token(token_key, token_secret)
+    auth = tweepy.OAuthHandler(config.get('consumer_key'), config.get('consumer_secret'))
+    auth.set_access_token(config.get('token_key'), config.get('token_secret'))
```
Comment (tklarryonline) [inline responsebot/utils/auth_utils.py]: 👍

File: responsebot/models.py (hunk 1 above)
Comment (giang271291) [inline responsebot/models.py]: @tklarryonline noted, thanks

**Pull request merged**